File: src/test/java/com/codingbat/warmup1/DiffTwentyOneTest.java
package com.codingbat.warmup1;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class DiffTwentyOneTest {
  @Test
  public void shouldEqualTwoWhenDiffingNineteen() {
    assertEquals(2, DiffTwentyOne.diff21(19));
  }

  @Test
  public void shouldEqualElevenWhenDiffingTen() {
    assertEquals(11, DiffTwentyOne.diff21(10));
  }

  @Test
  public void shouldEqualZeroWhenDiffingTwentyOne() {
    assertEquals(0, DiffTwentyOne.diff21(21));
  }

  @Test
  public void shouldEqualTwoWhenDiffingTwentyTwo() {
    assertEquals(2, DiffTwentyOne.diff21(22));
  }

  @Test
  public void shouldEqualEightWhenDiffingTwentyFive() {
    assertEquals(8, DiffTwentyOne.diff21(25));
  }

  @Test
  public void shouldEqualEighteenWhenDiffingThirty() {
    assertEquals(18, DiffTwentyOne.diff21(30));
  }

  @Test
  public void shouldEqualTwentyOneWhenDiffingZero() {
    assertEquals(21, DiffTwentyOne.diff21(0));
  }

  @Test
  public void shouldEqualTwentyWhenDiffingOne() {
    assertEquals(20, DiffTwentyOne.diff21(1));
  }

  @Test
  public void shouldEqualNineteenWhenDiffingTwo() {
    assertEquals(19, DiffTwentyOne.diff21(2));
  }

  @Test
  public void shouldEqualTwentyTwoWhenDiffingNegativeOne() {
    assertEquals(22, DiffTwentyOne.diff21(-1));
  }

  @Test
  public void shouldEqualTwentyThreeWhenDiffingNegativeTwo() {
    assertEquals(23, DiffTwentyOne.diff21(-2));
  }

  @Test
  public void shouldEqualFiftyEightWhenDiffingFifty() {
    assertEquals(58, DiffTwentyOne.diff21(50));
  }
}

File: src/test/kotlin/com/codingbat/warmup1/DiffTwentyOneTest.kt
package com.codingbat.warmup1

import org.junit.Test
import kotlin.test.assertEquals

class DiffTwentyOneTestKt {
  @Test
  fun shouldEqualTwoWhenDiffingNineteen() {
    assertEquals(2, diff21(19))
  }

  @Test
  fun shouldEqualElevenWhenDiffingTen() {
    assertEquals(11, diff21(10))
  }

  @Test
  fun shouldEqualZeroWhenDiffingTwentyOne() {
    assertEquals(0, diff21(21))
  }

  @Test
  fun shouldEqualTwoWhenDiffingTwentyTwo() {
    assertEquals(2, diff21(22))
  }

  @Test
  fun shouldEqualEightWhenDiffingTwentyFive() {
    assertEquals(8, diff21(25))
  }

  @Test
  fun shouldEqualEighteenWhenDiffingThirty() {
    assertEquals(18, diff21(30))
  }

  @Test
  fun shouldEqualTwentyOneWhenDiffingZero() {
    assertEquals(21, diff21(0))
  }

  @Test
  fun shouldEqualTwentyWhenDiffingOne() {
    assertEquals(20, diff21(1))
  }

  @Test
  fun shouldEqualNineteenWhenDiffingTwo() {
    assertEquals(19, diff21(2))
  }

  @Test
  fun shouldEqualTwentyTwoWhenDiffingNegativeOne() {
    assertEquals(22, diff21(-1))
  }

  @Test
  fun shouldEqualTwentyThreeWhenDiffingNegativeTwo() {
    assertEquals(23, diff21(-2))
  }

  @Test
  fun shouldEqualFiftyEightWhenDiffingFifty() {
    assertEquals(58, diff21(50))
  }
}
